### DISCUSSION TITLE
Are there ready-made solutions for displaying large texts in the likeness of man and more?

### DISCUSSION THREAD

**Discussion by KuzyaGlebkin** [varabyte/kotter#121] [Q&A]

**Comment by bitspittle** [comment]
Can you share a screenshot of what you are looking for? I don't know what you mean by "large texts". When I issue a `man` command, I see mostly regular text, with section headers that are ALL CAPS and **bolded**.

For example:

<img width="594" alt="Screenshot 2024-11-17 at 3 08 51 PM" src="https://github.com/user-attachments/assets/e20892ae-ce12-46f0-8803-5073b6fb97d0">

**Comment by KuzyaGlebkin** [comment]
I meant that the output of man is often large, which is why there is no automatic scrolling. I suppose that with this library, it is possible to try to implement this behavior by writing a custom wrapper. The wrapper would count the number of lines of output produced with the current number of columns, and then split the output into multiple screens based on the number of lines.

P.S.

I would like to apologize for the poor quality of my English. I am able to read in this language, however, the machine is currently writing for me.

**Reply by bitspittle** [reply]
No worries, having studied other languages even only a little, I can appreciate what you are doing here to communicate!

Even as a native speaker, sometimes I will share code and/or screenshots to make things really clear for other people... maybe that advice helps, maybe not! I've seen lots of hard to understand bugs even from very advanced English speakers.

I'm not too sure we can do what you want. I don't think you can query the height of the console at this point -- not sure how `man` does it honestly -- but I'm also not sure what you mean by "counting lines with columns and splitting onto multiple screens."

To scroll text in general, there is no pattern provided by Kotter, but it's not hard:

```kotlin
val lines: List<String> = ...
var start by liveVarOf(0)
section {
   for (i in start until 40) {
      textLine(lines[i])
   }
}.runUntilKeyPressed(Keys.Q) {
   onKeyPressed {
      when (key) {
         Keys.UP -> start--
         Keys.DOWN -> start++
      }
   } 
}
```

(Of course, you have to check the value of `start` to make sure it isn't less than zero or too large so that you go past the end of the `lines` value)

However, I'm realizing now that [offscreen buffers](https://github.com/varabyte/kotter?tab=readme-ov-file#-offscreen), which you might be wanting to use here, don't allow you to skip rendering rows, which might be useful for what you might be trying to do?
